### PR TITLE
NAS-132294 / 25.04 / Validate memory and add logs for failed start

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/virt/instance.py [hunk 1]
```diff
@@ -101,6 +101,14 @@ async def validate(self, new, schema_name, verrors, old=None):
         if not old and await self.query([('name', '=', new['name'])]):
             verrors.add(f'{schema_name}.name', f'Name {new["name"]!r} already exists')
 
+        if memory := new.get('memory'):
```
[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/virt/instance.py:104]: Thats a good idea, had not thought of that

[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/virt/instance.py]: Updated, thanks

File: src/middlewared/middlewared/api/v25_04_0/virt.py [hunk 2]
```diff
@@ -178,6 +178,13 @@ class VirtInstanceEntry(BaseModel):
     raw: dict
 
 
+# Lets require at least 32MiB of reserved memory
+# This value is somewhat arbitrary but hard to think lower value would have to be used
+# (would most likely be a typo).
+# Running container with very low memory will probably cause it to be killed by the cgroup OOM
+MemoryType: TypeAlias = Annotated[int, Field(strict=True, ge=33554432)]
```
Comment (Qubad786) [inline src/middlewared/middlewared/api/v25_04_0/virt.py:185]: You can use `conint` as well btw

**Review by Qubad786 — APPROVED** [review]
Thanks!

File: src/middlewared/middlewared/api/v25_04_0/virt.py (hunk 2 above)
Comment (william-gr) [reply src/middlewared/middlewared/api/v25_04_0/virt.py:185]: And will be deprecated, if I am reading correctly

**Pull request merged**